### PR TITLE
Fixed Pull Request for Mac OS updates.

### PR DESCRIPTION
**Pull request by danieljimenez**
1. Rename plist to proper name (broken in master)
	2. Add support for node['chef_client']['task']['user'].

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
 license           'Apache 2.0'
 description       'Manages client.rb configuration and chef-client service'
 long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '4.2.4'
+version           '4.2.42'
```
Comment (jtimberman) [inline metadata.rb]: Please don't change the version in the metadata.

File: attributes/default.rb [hunk 2]
```diff
@@ -127,11 +127,20 @@
 # don't use bsd paths per COOK-1379
 when 'mac_os_x', 'mac_os_x_server'
   default['chef_client']['init_style']  = 'launchd'
-  default['chef_client']['log_dir']     = '/Library/Logs/Chef'
-  # Launchd doesn't use pid files
-  default['chef_client']['run_path']    = '/var/run/chef'
-  default['chef_client']['cache_path']  = '/Library/Caches/Chef'
-  default['chef_client']['backup_path'] = '/Library/Caches/Chef/Backup'
+  
+  if node['chef_client']['task']['user'].eql? 'SYSTEM'
```
Comment (jtimberman) [inline attributes/default.rb]: This attribute is specifically for use with the Windows managed task, and I wouldn't rely on its value for OS X.

File: attributes/default.rb [hunk 3]
```diff
@@ -127,11 +127,20 @@
...
+    # Launchd doesn't use pid files
+    default['chef_client']['run_path']    = '/var/run/chef'
+    default['chef_client']['cache_path']  = '/Library/Caches/Chef'
+    default['chef_client']['backup_path'] = '/Library/Caches/Chef/Backup'
+  else
+    default['chef_client']['log_dir']     = "#{node['chef_client']['task']['user']}/Library/Logs/Chef"
+    # Launchd doesn't use pid files
+    default['chef_client']['run_path']    = '/var/run/chef'
+    default['chef_client']['cache_path']  = "#{node['chef_client']['task']['user']}/Library/Caches/Chef"
+    default['chef_client']['backup_path'] = "#{node['chef_client']['task']['user']}/Library/Caches/Chef/Backup"
```
Comment (jtimberman) [inline attributes/default.rb]: As noted before, `node['chef_client']['task']['user']` is for Windows task, and doesn't make sense to use in the OS X configuration.

**Comment by danieljimenez** [comment]
Again, suggestions welcome for where to store the user. Root is not appropriate to run chef as on Mac OS for most configurations.

**Comment by danieljimenez** [comment]
I guess I'll consider this a fork rather than a pull request then.

**Comment by jtimberman** [comment]
So here's the thing, in my opinion.

This is a cookbook that tries to do too much. It needs to be refactored to provide a resource, `chef_client_service` that has platform specific providers for managing a service in the "native" way for said platforms. It needs fewer node attributes. It needs to not be responsible for managing log rotation, or scheduled tasks/cron setup of chef-client runs. Likewise, there should be a `chef_client_config` resource that works similarly. 

Flexibility to run as a non-privileged user would be nice, but I think we should have the refactoring done before we approach that, as I think it can be done in a clean, cross platform way.

**Comment by chef-supermarket** [comment]
Hi. I am an automated pull request bot named Curry. There are commits in this pull request whose authors are not yet authorized to contribute to Chef Software, Inc. projects or are using a non-GitHub verified email address. To become authorized to contribute, you will need to sign the Contributor License Agreement (CLA) as an individual or on behalf of your company. [You can read more on Chef's blog.](https://www.chef.io/blog/2014/06/23/changes-to-the-contributor-license-agreement-process)

## GitHub Users Who Are Not Authorized To Contribute

The following GitHub users do not appear to have signed a CLA:

* @danieljimenez

[Please sign the CLA here.](https://supermarket.chef.io/become-a-contributor)

**Comment by jtimberman** [comment]
I think this may conflict with #294's changes and not merge cleanly.

**Comment by juliandunn** [comment]
@danieljimenez are you still working on this?

**Comment by juliandunn** [comment]
No response in some time so I assume OP is no longer interested in this.

**Comment by danieljimenez** [comment]
I am interested but unable to find a working solution or everyone.

**Pull request closed without merging**